datatypes/src/primitives/feature_data.rs

```diff
@@ -3,370 +3,260 @@ use crate::primitives::PrimitivesError;
 use crate::util::Result;
 use arrow::bitmap::Bitmap;
 use serde::{Deserialize, Serialize};
+use serde_json::Value;
 use snafu::ensure;
 use std::convert::TryFrom;
 use std::slice;
 use std::str;
 
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Deserialize, Serialize)]
 pub enum FeatureDataType {
```

Comment:
I am wondering why you didn't have to change the `FeatureCollectionBuilder` at all. Can I still fix the nullability of a column at the start? I think there exists no test that actually adds a nullable column.

Reply:
Why should we fix the nullability of a column at the start?

Reply:
because of data integrity. NOT NULL constraints are very common in SQL